presto-main/src/main/java/com/facebook/presto/execution/scheduler/ResettableRandomizedIterator.java

```diff
@@ -36,6 +36,11 @@ public void reset()
         position = 0;
     }
 
+    public int size()
```

Comment:
It's kind of weird for an iterator to have `size`. Plus, this looks really like an `ListIterator` :)

Reply:
Yeah, not nice. But the class is pretty private. I would simply keep it. The usages are pretty readable (e.g: `candidates.size()`)